openedx/features/calendar_sync/plugins.py

```diff
@@ -0,0 +1,74 @@
+"""
+Platform plugins to support Calendar Sync toggle.
+"""
+
+
+from django.urls import reverse
+from django.utils.translation import ugettext as _
+
+from openedx.features.calendar_sync.api import SUBSCRIBE, UNSUBSCRIBE
+from openedx.features.calendar_sync.models import UserCalendarSyncConfig
+from openedx.features.course_experience import CALENDAR_SYNC_FLAG, RELATIVE_DATES_FLAG
+from openedx.features.course_experience.course_tools import CourseTool, HttpMethod
+from student.models import CourseEnrollment
+
+
+class CalendarSyncToggleTool(CourseTool):
+    """
+    The Calendar Sync toggle tool.
+    """
+    http_method = HttpMethod.POST
+    link_title = _('Calendar Sync')
+    toggle_data = {'toggle_data': ''}
+
+    @classmethod
+    def analytics_id(cls):
+        """
+        Returns an id to uniquely identify this tool in analytics events.
+        """
+        return 'edx.calendar-sync'
+
+    @classmethod
+    def is_enabled(cls, request, course_key):
+        """
+        The Calendar Sync toggle tool is limited to user enabled through a waffle flag.
+        Staff always has access.
+        """
+        if CALENDAR_SYNC_FLAG.is_enabled(course_key) and RELATIVE_DATES_FLAG.is_enabled(course_key):
```

Comment:
Super minor nit: it might make the eventual PR dropping these lines to be smaller if you did an early exit rather than an if-wrapper.